sql/mysql_db/mysql_db.go

```diff
@@ -30,9 +30,26 @@ import (
 	"github.com/dolthub/go-mysql-server/sql/mysql_db/serial"
 )
 
-// PersistCallback represents the callback that will be called when the Grant Tables have been updated and need to be
-// persisted.
-type PersistCallback func(ctx *sql.Context, data []byte) error
+// MySQLDbPersistence is used to determine the behavior of how certain tables in MySQLDb will be persisted.
+type MySQLDbPersistence interface {
+	CanPersist() bool
```

Comment:
Make this return an error instead, call it `ValidateCanPersist()`

Then integrators can return their own error message if they can't persist the data